Jobs/PrGate.yml

```diff
@@ -68,64 +64,94 @@ parameters:
   displayName: Virtual Machine Image (e.g. windows-latest)
   type: string
   default: 'windows-latest'
+- name: extra_install_step
+  displayName: Extra Install Steps
+  type: stepList
+  default: []
+- name: pool_name
+  displayName: Variable name that hosts pool name to be used for self-hosted agents
+  type: string
+  default: ''
+- name: os_type
+  displayName: OS type on the self-hosted agent pools
+  type: string
+  values:
+  - Windows_NT
+  - Linux
+  default: Windows_NT
+- name: build_matrix
+  type: object
+  # Each element in the matrix will be used in a separate job
+  # Required fields:
+  # Pkgs:           Specify what packages in the target repo are to be build for this job
+  # Targets:        Specify which targets are to be built in this job, i.e. DEBUG,
+  #                 RELEASE, NO-TARGET, NOOPT. The acceptable values depend on the
+  #                 Pkgs to be built.
+  # ArchList:       List of architectures to be supported by the packages (e.g. IA32, X64)
+  # SelfHostAgent:  A boolean indicating whether this job should be run on the selfhosted
+  #                 "pool_name"
 
 # Build step
 jobs:
 
-  - template: ../Jobs/CreateBuildMatrix.yml
-    parameters:
-      group_package_list: ${{ parameters.packages }}
-      target_list: ${{ parameters.target_list }}
-
-  - job: Build
-    dependsOn: CreateBuildMatrix
-    condition: Succeeded('CreateBuildMatrix')
-
-    # Use matrix to speed up the build process
-    strategy:
-      matrix: $[ dependencies.CreateBuildMatrix.outputs['CalculateMatrix.Matrix'] ]
+- ${{ each item in parameters.build_matrix }}:
+  - job: Build_${{ item.Key }}
+    timeoutInMinutes: 120
+    condition: |
+      and(not(Canceled()),
+          or(${{ eq(item.Value.SelfHostAgent, false) }}, ne(variables['${{ parameters.pool_name }}'], '')))
 
     workspace:
       clean: all
 
-    pool:
-      vmImage: ${{ parameters.vm_image }}
+    ${{ if eq(item.Value.SelfHostAgent, true) }}:
+      pool:
+        name: $(${{ parameters.pool_name }})
+        demands:
+        - Agent.OS -equals ${{ parameters.os_type }}
+    ${{ else }}:
+      pool:
+        vmImage: ${{ parameters.vm_image }}
 
     # Use a container if one was specified.
-    ${{ if and(eq(parameters.container_build, true), not(contains(parameters.vm_image, 'windows'))) }}:
+    ${{ if and(eq(parameters.container_build, true), not(contains(parameters.vm_image, 'windows')), not(eq(item.Value.SelfHostAgent, true))) }}:

```

Comment:
nit: consider using ne() instead of not(eq())

Reply:
below as well

Reply:
Agreed, updated.